src/us/kbase/workspace/database/exceptions/ErrorOr.java

```diff
@@ -0,0 +1,96 @@
+package us.kbase.workspace.database.exceptions;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+/** 
+ * A class that represents an object or an error that occurred attempting to create or retrieve
+ * the object.
+ */
+public class ErrorOr<T> {
+
+	// TODO ERROR may want to support including custom error fields? Different errors = different fields
+	// TODO LOMBOK
```

Comment:
Lombok = https://projectlombok.org/ ?

Reply:
correct

Reply:
I'd like to move TODOs out of the code into tickets.

Reply:
It seems you and I are pretty far apart on how we treat TODOs. I'm happy to have that discussion, but not in a PR review - we should probably move it to slack or something.

Reply:
We can discuss over a video chat.